include/zephyr/drivers/clock_control/stm32_clock_control.h

```diff
@@ -16,6 +16,8 @@
 #include <zephyr/dt-bindings/clock/stm32c0_clock.h>
 #elif defined(CONFIG_SOC_SERIES_STM32F0X)
 #include <zephyr/dt-bindings/clock/stm32f0_clock.h>
+#elif defined(SOC_STM32F10X_CONNECTIVITY_LINE_DEVICE)
```

Comment:
Nit: it could be wise to document that this check *must* be done before the `if defined(CONFIG_SOC_SERIES_STM32F1X)` check.